examples/ImGuiModule.cmake

```diff
@@ -1,4 +1,7 @@
-include_guard()
+if(__CURRENT_FILE_VAR__)
+  return()
+endif()
+set(__CURRENT_FILE_VAR__ TRUE)
```

Comment:
Thank you for your efforts, but it seems that the code above though works, but does not guarantee the expected behavior.
It seems to me that you just copied a piece of code from the CMake documentation, but did not adapt it to a specific case.